keras/models.py

```diff
@@ -139,7 +139,8 @@ def _clone_functional_model(model, input_tensors=None):
                         layer(computed_tensor, **kwargs))
                     output_masks = to_list(
                         layer.compute_mask(computed_tensor,
-                                           computed_mask))
+                                           computed_mask)) \
+                        if layer.supports_masking else [None] * len(output_tensors)
```

Comment:
This should be structured as an `if` block for better readability:

```
if ...:
   output_masks = ...
else:
   output_masks = ...
```

You can simply specify a lack of mask by passing `None` (it doesn't have to be a list of None)

Reply:
Thanks for pointing that out! I agree with you that using `if` blocks is better and I'll commit it soon. 
But I think to assign a list of `None` is necessary, because `output_masks` will be passed to `zip` as the second argument later, so in my opinion it should be an iterable object but not a `None` (otherwise it can cause an error).